models/cartDao.js

```diff
@@ -0,0 +1,55 @@
+const appDataSource = require('./appDataSource');
+
+const createOrUpateCart = async (productId, userId) => {
+  return await appDataSource.query(
+    `INSERT INTO carts(
+            product_id,
+            user_id,
+            quantity
+        ) VALUES (?, ?, 1)
+        ON DUPLICATE KEY UPDATE
+        quantity = quantity + 1`,
+    [productId, userId]
+  );
+};
+
+const deleteFromCart = async (cartId) => {
+  return await appDataSource.query(
+    `DELETE FROM carts
+    WHERE id IN (?)`,
+    [cartId]
+  );
+};
```

Comment:
```suggestion
const deleteFromCart = async (cartId, userId) => {
  return await appDataSource.query(
    `DELETE FROM carts
    WHERE id IN (?) AND user_id = ?`,
    [ cartId, userId ]
  );
};
```
carts의 pk 값 만으로도 실제 삭제시키는데는 문제없겠지만, 혹여 query string으로 fetch 할 때, 토큰의 주인과 관련 없는 다른 id 값으로 요청이 들어오게된다면 문제가 있을 수 있습니다. 따라서 인가코드를 거치며 검증된 userId의 조건을 추가해주면 보다 보안적으로 안정성 있는 delete query가 될 수 있습니다.